### PR TITLE
fix(logging): improve python core's debug logging

### PR DESCRIPTION
**Pull request by padamstx**

### CODE REVIEW

File: ibm_cloud_sdk_core/authenticators/bearer_token_authenticator.py [hunk 1]
```diff
@@ -15,9 +15,11 @@
 # limitations under the License.
 
 from requests import Request
-
```
Comment (pyrooka) [inline ibm_cloud_sdk_core/authenticators/bearer_token_authenticator.py:18]: This empty line should be there to separate 3rd party and local package imports. :)

File: ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py [hunk 2]
```diff
@@ -15,11 +15,13 @@
 # limitations under the License.
 
 from typing import Dict
-
```
Comment (pyrooka) [inline ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py:18]: This empty line should be there to separate std lib and 3rd party package imports. :)

**Review by pyrooka — COMMENTED** [review]
I think there should be 2 empty lines between the imports and the `logger` declaration. IIRC top level code blocks should be surrounded by 2 empty lines. Feel free to correct me if I'm wrong!

**Review by pyrooka — APPROVED** [review]
LGTM!

File: ibm_cloud_sdk_core/authenticators/bearer_token_authenticator.py (hunk 1 above)
Comment (padamstx) [reply ibm_cloud_sdk_core/authenticators/bearer_token_authenticator.py:18]: fixed

File: ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py (hunk 2 above)
Comment (padamstx) [reply ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py:18]: fixed

File: test/utils/logger_utils.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import logging
+from http.client import HTTPConnection
+from ibm_cloud_sdk_core.logger import get_logger
+
+
+def setup_test_logger(level: int) -> logging.Logger:
+    """Sets up logging with a console handler to assist test cases."""
+    logger = get_logger()
+    logger.setLevel(level)
+    # if not logger.hasHandlers():
```
Comment (dpopp07) [inline test/utils/logger_utils.py:26]: Is this comment outdated?

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: test/utils/logger_utils.py (hunk 3 above)
Comment (padamstx) [reply test/utils/logger_utils.py:26]: probably :)   I'll fix it before merging.

**Comment by pyrooka** [comment]
As I'm looking at the changes and the README. Shouldn't we update the logging section of the README? Especially the code below. Now it's enough to enable the debug logging before the base service is created, right?
```markdown
### Low level request and response dump
To get low level information of the requests/ responses:

from http.client import HTTPConnection
HTTPConnection.debuglevel = 1
```

**Comment by padamstx** [comment]
> As I'm looking at the changes and the README. Shouldn't we update the logging section of the README? Especially the code below. Now it's enough to enable the debug logging before the base service is created, right?
> 
> ```
> ### Low level request and response dump
> To get low level information of the requests/ responses:
> 
> from http.client import HTTPConnection
> HTTPConnection.debuglevel = 1
> ```

Yes, I'm planning on doing that prior to merging.

**Comment by padamstx** [comment]
@dpopp07 @pyrooka I think I can say that this PR is finally finished! :joy:  In my more recent changes, I've simplified a bit how testcases can enable debug logging, and I also updated the README.md to have more up-to-date info about logging.

**Review by pyrooka — APPROVED** [review]
I think I went through all the changes that you made since my last review, and I still think this looks good! :)

**Review by dpopp07 — APPROVED** [review]
New changes look good! 👍

**Comment by dpopp07** [comment]
Looks pretty similar so maybe it's good to go already but does this have any implications for how the _SDKs_ do logging, as documented [here](https://github.com/IBM/ibm-cloud-sdk-common?tab=readme-ov-file#logging)?

**Comment by padamstx** [comment]
> Looks pretty similar so maybe it's good to go already but does this have any implications for how the _SDKs_ do logging, as documented [here](https://github.com/IBM/ibm-cloud-sdk-common?tab=readme-ov-file#logging)?

Yes, I have a to-do item to go through the logging-related info in the common sdk repo and make adjustments as needed after all the logging-related changes in the cores.

**Pull request merged**